main/ejml-dsparse/src/org/ejml/sparse/csc/CommonOps_DSCC.java

```diff
@@ -1835,5 +1837,111 @@ public static double trace( DMatrixSparseCSC A ) {
 
         return output;
     }
+
+    /**
+     * This applies a given unary function on every value stored in the matrix
+     *
+     * @param input  (Input) input matrix. Not modified
+     * @param func   Unary function accepting a double
+     * @param output (Input/Output) Matrix. Modified.
+     */
+    public static void apply(DMatrixSparseCSC input, DoubleUnaryOperator func, @Nullable DMatrixSparseCSC output) {
+        if (output == null) {
+            output = input.createLike();
+        } else if (input != output) {
+            output.copyStructure(input);
+        }
+
+        for (int i = 0; i < input.nz_length; i++) {
+            output.nz_values[i] = func.apply(input.nz_values[i]);
+        }
+    }
+
+    public static void apply(DMatrixSparseCSC input, DoubleUnaryOperator func) {
+        apply(input, func, input);
+    }
+
+    /**
+     * This accumulates the matrix values to a scalar value
+     * @param input (Input) input matrix. Not modified
+     * @param initValue initial value for accumulator
+     * @param func Accumulator function defining "+" for accumulator +=  cellValue
+     * @return accumulated value
+     */
+    public static double reduceScalar(DMatrixSparseCSC input, double initValue, DoubleBinaryOperator func) {
+        double result = initValue;
+
+        for (int i = 0; i < input.nz_length; i++) {
+            result = func.apply(result, input.nz_values[i]);
+        }
+
+        return result;
+    }
+
+    public static double reduceScalar(DMatrixSparseCSC input, DoubleBinaryOperator func) {
+        return reduceScalar(input, 0, func);
+    }
+
+    /**
+     * This accumulates the values per column to a scalar value
```

Comment:
same comment here about adding matlab like code to explain what reduce operation does